scipy/linalg/tests/test_lapack.py

```diff
@@ -1645,3 +1645,50 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def test_gttrf_gttrs():
+    np.random.seed(42)
+    n = 10
+    for index, dtype in enumerate(DTYPES):
+        atol = 10**-(np.finfo(dtype).precision-1)
+        if index < 2:
+            du = np.random.rand(n-1)
+            d = np.random.rand(n)
+            dl = np.random.rand(n-1)
+        else:
+            du = np.random.rand(n-1) * 1j
+            d = np.random.rand(n) * 1j
+            dl = np.random.rand(n-1) * 1j
+
+        diag_cpy = np.array([dl.copy(), d.copy(), du.copy()])
+
+        A = np.diag(d) + np.diag(dl, -1) + np.diag(du, 1)
+        x = np.random.rand(n)
+        b = A @ x
+
+        dgttrf = get_lapack_funcs('gttrf', dtype=dtype)
+        dgttrs = get_lapack_funcs('gttrs', dtype=dtype)
+
+        _dl, _d, _du, du2, ipiv, info = dgttrf(dl, d, du)
+
+        np.testing.assert_allclose(dl, diag_cpy[0], atol=atol)
```

Comment:
I would `assert_array_equal` here, right? We want to verify that they haven't changed _at all_.